keep/contextmanager/contextmanager.py

```diff
@@ -187,4 +195,38 @@ def load_step_context(self, step_id, step_results, step_conditions):
 
     # TODO - add step per alert?
     def get_step_context(self, step_id):
-        return {"step_id": step_id, "step_context": self._steps_context.get(step_id)}
+        return {"step_id": step_id, "step_context": self.steps_context.get(step_id)}
+
+    def __load_state(self):
+        if self.state_file:
+            # TODO - SQLite
+            try:
+                with open(self.state_file, "r") as f:
+                    self.state = json.load(f)
+            except:
+                self.state = {}
+
+    def get_last_alert_run(self, alert_id):
+        if alert_id in self.state:
+            return self.state[alert_id][-1]
+        # no previous runs
```

Comment:
should it fail?

Reply:
idk, let's talk about it, you defined this one, I just moved it